packages/connect-web/src/index.ts

```diff
@@ -1,107 +1,45 @@
+import { TrezorConnectDynamic } from '@trezor/connect/src/impl/dynamic';
 import { CoreInIframe } from './impl/core-in-iframe';
 import { CoreInPopup } from './impl/core-in-popup';
-import ProxyEventEmitter from './utils/proxy-event-emitter';
-import type { ConnectSettings, ConnectSettingsPublic, Manifest } from '@trezor/connect/src/types';
-import EventEmitter from 'events';
-import { CallMethodPayload } from '@trezor/connect/src/events';
+import type { ConnectSettingsPublic } from '@trezor/connect';
 import { getEnv } from './connectSettings';
 
-type TrezorConnectType = 'core-in-popup' | 'iframe';
-
 const IFRAME_ERRORS = ['Init_IframeBlocked', 'Init_IframeTimeout', 'Transport_Missing'];
 
-/**
- * Implementation of TrezorConnect that can dynamically switch between iframe and core-in-popup implementations
- */
-export class TrezorConnectDynamicImpl implements ConnectFactoryDependencies {
-    public eventEmitter: EventEmitter;
-
-    private currentTarget: TrezorConnectType = 'iframe';
-    private coreInIframeImpl: CoreInIframe;
-    private coreInPopupImpl: CoreInPopup;
-
-    private lastSettings?: Partial<ConnectSettings>;
-
-    public constructor() {
-        this.coreInIframeImpl = new CoreInIframe();
-        this.coreInPopupImpl = new CoreInPopup();
-        this.eventEmitter = new ProxyEventEmitter([
-            this.coreInIframeImpl.eventEmitter,
-            this.coreInPopupImpl.eventEmitter,
-        ]);
-    }
-
-    private getTarget() {
-        return this.currentTarget === 'iframe' ? this.coreInIframeImpl : this.coreInPopupImpl;
-    }
-
-    private async switchTarget(target: TrezorConnectType) {
-        if (this.currentTarget === target) {
-            return;
-        }
-
-        await this.getTarget().dispose();
-        this.currentTarget = target;
-        await this.getTarget().init(this.lastSettings);
-    }
-
-    public manifest(manifest: Manifest) {
-        this.lastSettings = {
-            ...this.lastSettings,
-            manifest,
-        };
-
-        this.getTarget().manifest(manifest);
-    }
-
-    public async init(settings: Partial<ConnectSettingsPublic> = {}) {
+const impl = new TrezorConnectDynamic<'core-in-popup' | 'iframe'>({
+    implementations: [
+        {
+            type: 'iframe',
+            impl: new CoreInIframe(),
+        },
+        {
+            type: 'core-in-popup',
+            impl: new CoreInPopup(),
+        },
+    ],
+    getInitTarget: (settings: Partial<ConnectSettingsPublic>) => {
         const env = getEnv();
         if (settings.coreMode === 'iframe' || settings.popup === false || env === 'webextension') {
-            this.currentTarget = 'iframe';
+            return 'iframe';
         } else if (settings.coreMode === 'popup') {
-            this.currentTarget = 'core-in-popup';
+            return 'core-in-popup';
         } else {
+            // TODO: it does not look like a good idea to modify settings like this, create a public method in dynamic for that?
```

Comment:
Yeah, the current code probably can't work

Reply:
I think it actually works because JS objects are copy by reference. But it does not look nice.
I am thinking is it is actually needed.

Reply:
If you know why it is required we can keep doing it by adding a public method to TrezorConnectDynamic like:

```typescript
    public updateSettings(settingToUpdate: Partial<ConnectSettingsPublic>) {}
```